packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -240,26 +280,29 @@ export function useInputHandler<T extends ValidatableFormElement>(
         // (We're also expecting the use of radio buttons with our default validation message handling
         // to be fairly uncommon because the overall visual experience is clunky with such a configuration.)
         const onAttributeChange = (): void => {
+            updateCustomValidationMessage();
             if (!isValid.value) checkHtml5Validity();
         };
         let validationAttributeObserver: MutationObserver | null = null;
         watch(
-            [maybeElement, isValid, () => props.useHtml5Validation],
+            [maybeElement, (): boolean => props.useHtml5Validation ?? true],
             (data) => {
                 // Not using destructuring assignment because browser support is just a little too weak at the moment
                 const el = data[0];
-                const valid = data[1];
-                const useValidation = data[2];
-
-                // Clean up previous state.
-                if (validationAttributeObserver != null) {
-                    // Process any pending events.
-                    if (validationAttributeObserver.takeRecords().length > 0)
-                        onAttributeChange();
-                    validationAttributeObserver.disconnect();
-                }
+                const useValidation = data[1];
 
-                if (!isDefined(el) || valid || !useValidation) return;
+                if (!isDefined(el) || !useValidation) {
+                    // Clean up previous state.
+                    if (validationAttributeObserver != null) {
```

Comment:
Why move the cleanup inside this `if`? 
I don't know exactly how the browser handles old objects, but now every time we create a `new MutationObserver` we override the old object and don't disconnect the old element. This could be a memmory leak problem until the old instance is cleaned up by the browser, which is out of our hands.

Reply:
The idea here was that we now need to run the observer even when the input is valid. If we always update the validation message in `checkHtml5Validity`, though, that may no longer be necessary.

We only construct a new `MutationObserver` if `validationAttributeObserver` is `null`, which only happens when the variable is first initialized. We do disconnect and reconnect the observer, but that should actually work as expected in most cases. The first connection [will automatically disconnect all existing connections if the element hasn't changed](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#reusing_mutationobservers). If the element has changed, though, this code would have issues.

The next version I'm working on probably won't need this change anyway, but if it does, there's a little rework I'll want to do here.

Reply:
Unfortunately, it does look like there are cases where this watcher has to get involved for valid inputs. I've cleaned this section up a bit, though.